src/components/Product/Product.js

```diff
@@ -1,43 +1,87 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import firebase from '@firebase/app';
+
 function Product() {
   const [option, setOption] = useState('');
   const [data, setData] = useState({
     name: '',
     lastDate: null,
   });
-  const addProduct = (event) => {
-    event.preventDefault();
-    console.log('send data...' + data.name, data.lastDate, option);
+
+  const [error, setError] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [productData, setProductData] = useState([]);
+
+  const viewMessage = (message, error) => {
+    setErrorMessage(message);
+    setError(error);
+  };
+
+  useEffect(() => {
     firebase
       .firestore()
-      .collection(token())
+      .collection('products')
+      .onSnapshot((snapshot) => {
+        let products = [];
+        snapshot.forEach((doc) => products.push(doc.data().name));
+        setProductData(products);
+      });
+  }, []);
+
+  const validateDuplicate = (product) => {
+    let productDataResult = productData;
+    if (productDataResult.indexOf(product) > -1) {
+      return true;
+    }
+    return false;
+  };
+
+  const addProduct = (colecction, nameProduct) => {
+    colecction
       .add({
-        name: data.name,
+        name: nameProduct,
         lastDate: data.lastDate,
         date: option,
       })
-      .then((quantityAfter) => {
-        console.log('the amount was saved successfully', quantityAfter);
+      .then(() => {
+        viewMessage('Successfully Added', 'success');
       })
       .catch((error) => {
-        console.log('error', error);
+        viewMessage('Error', error);
       });
   };
+
+  const validate = async (event) => {
+    event.preventDefault();
+    const colecction = firebase.firestore().collection('products');
+    const nameProduct = formate(data.name);
+    try {
+      const isDuplicate = validateDuplicate(nameProduct);
+      if (isDuplicate) {
+        viewMessage(`The product: ${data.name} already exists!!`, 'error');
+      } else {
+        addProduct(colecction, nameProduct);
+      }
+    } catch (e) {
+      viewMessage('Error', error);
+    }
+  };
+
   const handleInputChange = (event) => {
     setData({
       ...data,
       [event.target.name]: event.target.value,
     });
   };
-  var rand = function () {
-    return Math.random().toString(36).substr(2);
-  };
-  var token = function () {
-    return rand() + rand();
+
+  const formate = (name) => {
```

Comment:
I think you misspelled this work, it should be `format` not `formate` 😬

Reply:
jejejejej is corrected !!